### PR TITLE
Update to use `purchases-hybrid-common` 3.2.2

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes [CSDK-313]

### Changes:
- Removed `alias`/`reset`/`identify`
- Renamed `PurchaserInfo` to `CustomerInfo`
- Removed deprecated `addAttributionData`
- Renamed `getPaymentDiscount` to `getPromotionalOffer`
- Renamed `PurchasesPaymentDiscount` to `PurchasesPromotionalOffer`
- Renamed `restorePurchases` to `restoreTransactions`

[CSDK-313]: https://revenuecats.atlassian.net/browse/CSDK-313?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ios/RNPurchases.m [hunk 1]
```diff
@@ -209,19 +186,20 @@ - (dispatch_queue_t)methodQueue {
...
-                 getPaymentDiscountForProductIdentifier:(NSString *)productIdentifier
-                 discountIdentifier:(nullable NSString *)discountIdentifier
-                 resolve:(RCTPromiseResolveBlock)resolve
-                 reject:(RCTPromiseRejectBlock)reject) {
-    [RCCommonFunctionality paymentDiscountForProductIdentifier:productIdentifier
-                                                      discount:discountIdentifier
-                                               completionBlock:[self getResponseCompletionBlockWithResolve:resolve
-                                                                                                    reject:reject]];
-}
+// TODO: ?
```
Comment (NachoSoto) [inline ios/RNPurchases.m:189]: I need to fix this

File: android/gradle.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 android.useAndroidX=true
-Purchases_kotlinVersion=1.4.0
+Purchases_kotlinVersion=1.5.31
```
Comment (NachoSoto) [inline android/gradle.properties:2]: This is required now by [RN 0.67](https://reactnative.dev/blog/2022/01/19/version-067).

File: ios/RNPurchases.m [hunk 3]
```diff
@@ -209,19 +184,8 @@ - (dispatch_queue_t)methodQueue {
     }];
 }
 
-RCT_REMAP_METHOD(getPaymentDiscount,
```
Comment (NachoSoto) [inline ios/RNPurchases.m:212]: Looks like this is gone?

File: android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java [hunk 4]
```diff
@@ -82,23 +81,14 @@ public void setupPurchases(String apiKey, @Nullable String appUserID,
...
+        Purchases.getSharedInstance().setUpdatedCustomerInfoListener(this);
     }
 
     @ReactMethod
     public void setAllowSharingStoreAccount(boolean allowSharingStoreAccount) {
         CommonKt.setAllowSharingAppStoreAccount(allowSharingStoreAccount);
     }
 
-    @ReactMethod
-    public void addAttributionData(ReadableMap data, Integer network, @Nullable String networkUserId) {
```
Comment (NachoSoto) [inline android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java:94]: Looks like this is gone too?

Comment (aboedo) [reply android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java:94]: yes

File: ios/RNPurchases.m (hunk 3 above)
Comment (aboedo) [reply ios/RNPurchases.m:212]: this one was replaced by getPromoOffer

File: android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java (hunk 4 above)
Comment (NachoSoto) [reply android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java:94]: I'll remove it from iOS too then.

File: CHANGELOG.md [hunk 5]
```diff
@@ -212,9 +212,9 @@ Detailed docs about the new system are available [here](https://docs.revenuecat.
 
 - Removes usage of BuildConfig [#171](https://github.com/RevenueCat/react-native-purchases/pull/171)
 - Bumped common files to 1.4.2 [Changelog here](https://github.com/RevenueCat/purchases-hybrid-common/releases/tag/1.4.2)
-- Bumped iOS to 3.7.1 [Changelog here](https://github.com/RevenueCat/purchases-ios/releases/tag/3.7.1) 
+- Bumped iOS to 3.7.1 [Changelog here](https://github.com/RevenueCat/purchases-ios/releases/tag/3.7.1)
 - Bumped Android to 3.5.0 [Changelog here](https://github.com/RevenueCat/purchases-android/releases/tag/3.5.0)
-- Added a new property `nonSubscriptionTransactions` in `PurchaserInfo` to better manage non-subscriptions
+- Added a new property `nonSubscriptionTransactions` in `CustomerInfo` to better manage non-subscriptions
```
Comment (aboedo) [inline CHANGELOG.md:217]: for old versions I think we should keep the old nomenclature

File: android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java (hunk 4 above)
Comment (NachoSoto) [reply android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java:94]: https://github.com/RevenueCat/purchases-hybrid-common/pull/169

File: ios/RNPurchases.m (hunk 3 above)
Comment (NachoSoto) [reply ios/RNPurchases.m:212]: Oh that's right, fixing

File: ios/RNPurchases.m [hunk 6]
```diff
@@ -329,17 +293,17 @@ - (dispatch_queue_t)methodQueue {
...
+- (void)purchases:(RCPurchases *)purchases didReceiveUpdatedCustomerInfo:(RCCustomerInfo *)customerInfo {
+    [self sendEventWithName:RNPurchasesCustomerInfoUpdatedEvent body:customerInfo.dictionary];
 }
 
-- (void)         purchases:(RCPurchases *)purchases
-shouldPurchasePromoProduct:(SKProduct *)product
-            defermentBlock:(RCDeferredPromotionalPurchaseBlock)makeDeferredPurchase {
+- (void)purchases:(RCPurchases *)purchases
+readyForPromotedProduct:(RCStoreProduct *)product
+         purchase:(void (^)(void (^ _Nonnull)(RCStoreTransaction * _Nullable, RCCustomerInfo * _Nullable, NSError * _Nullable, BOOL)))startPurchase {
```
Comment (aboedo) [inline ios/RNPurchases.m:293]: we should update the indentation here for consistency

File: src/purchases.ts [hunk 7]
```diff
@@ -105,9 +105,9 @@ export enum BILLING_FEATURE {
  */
 export interface LogInResult {
   /**
-   * The Purchaser Info for the user.
```
Comment (aboedo) [inline src/purchases.ts:108]: addAttributionData should be removed from this file too, right?

File: CHANGELOG.md (hunk 5 above)
Comment (NachoSoto) [reply CHANGELOG.md:217]: Oops this was find-replace being too broad.

File: ios/RNPurchases.m (hunk 6 above)
Comment (NachoSoto) [reply ios/RNPurchases.m:293]: How so? The second parameter doesn't fit. Or do you mean something else?

File: src/purchases.ts (hunk 7 above)
Comment (NachoSoto) [reply src/purchases.ts:108]: Good catch.

File: ios/RNPurchases.m (hunk 3 above)
Comment (NachoSoto) [reply ios/RNPurchases.m:212]: Fixed.

File: VERSIONS.md [hunk 8]
```diff
@@ -1,5 +1,6 @@
 | Version | iOS version | Android version | Common files version |
 |---------|-------------|-----------------|----------------------|
+| 5.0.0   | 4.7.0       | 5.1.0           | 3.2.1                |
```
Comment (NachoSoto) [inline VERSIONS.md:3]: I'll update this in a future PR before #367

File: README.md [hunk 9]
```diff
@@ -57,14 +57,14 @@ In your `ios` folder, run `pod install`. If you've just upgraded ExpoKit, you mi
 
 ##### Make a Framework Reference in your project
 
-1. Drag `Purchases.framework` and `PurchasesHybridCommon.framework` from the `RNPurchases`sub-project under the libraries section to the outer project and create a reference.
```
Comment (aboedo) [inline README.md:60]: hopefully this entire process gets deleted soon, but that's a conversation on a separate PR (https://github.com/RevenueCat/react-native-purchases/pull/323) and shouldn't affect this one

File: VERSIONS.md (hunk 8 above)
Comment (aboedo) [inline VERSIONS.md:3]: 3.2.2, right? I think we also need to update the Android version number

Comment (aboedo) [reply VERSIONS.md:3]: we can do that as a separate thing though since 3.2.3 is already on the way

File: ios/RNPurchases.m [hunk 10]
```diff
@@ -112,7 +104,7 @@ - (dispatch_queue_t)methodQueue {
 RCT_REMAP_METHOD(restoreTransactions,
                  restoreTransactionsWithResolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject) {
-    [RCCommonFunctionality restoreTransactionsWithCompletionBlock:[self getResponseCompletionBlockWithResolve:resolve
+    [RCCommonFunctionality restorePurchasesWithCompletionBlock:[self getResponseCompletionBlockWithResolve:resolve
                                                                                                        reject:reject]];
```
Comment (aboedo) [inline ios/RNPurchases.m:108]: indentation here is off

**Review by aboedo — APPROVED** [review]

File: VERSIONS.md (hunk 8 above)
Comment (NachoSoto) [reply VERSIONS.md:3]: Oh that's right. I'll update this one for now for consistency.

**Pull request merged**